server-spi/src/main/java/org/keycloak/http/HttpResponse.java

```diff
@@ -45,4 +46,20 @@ public interface HttpResponse {
      * @param value the header value
      */
     void setHeader(String name, String value);
+
+    /**
+     * Sets a new cookie only if not yet set.
+     *
+     * @param cookie the cookie
+     */
+    void setCookieIfAbsent(HttpCookie cookie);
```

Comment:
@pedroigor Do we really need to "pollute" the `HttpResponse` interface with the custom class `HttpCookie` only due to the presence of the method converting the cookie props to the Set-Cookie form attribute? We could do it additionally in our implementation of the interface, right? 

Moreover, I'd rather see in this interface some standard structure such as `NewCookie` which should handle the whole process with Set-Cookie stuff.

For inspiration: https://github.com/mabartos/keycloak/commit/89627088421c6c0d2fd60a6a0cade64f8e159a77

Reply:
"`NewCookie`, when sent in the Response, will set a Set-Cookie response header with the cookie information, and `Cookie` will set the Cookie request header with the cookie information. This is per the HTTP spec." [1]

[1] https://stackoverflow.com/a/34055702

Reply:
How are you handling SameSite cookies in that branch?

Regarding the changes to `HttpResponse` and `HttpCookie`, I'm trying to be a bit conservative here and keep the same behavior when setting cookies. Removing/deprecating code or calls from/to `ServerCookie` should be done carefully as JAX-RS does not provide support for samesite cookies.

Or does it?

Reply:
TBH, I prefer having the same representation/type for both request and response cookies.

Reply:
>How are you handling SameSite cookies in that branch?

It's not a complete solution for that yet. I just wanted to mention a particular approach exists. 

>TBH, I prefer having the same representation/type for both request and response cookies.

Not sure if it has some benefits. As the standard way of using cookies is to use `NewCookie` (extension of Cookie) for response and `Cookie` for request; we can easily obtain `Cookie` from `NewCookie`.

However, we can discuss it in different channels and it can be done in some follow-up tasks as this may be slightly out of the scope of this PR and requires some investigation around it. As the next major version of Keycloak is far away, we have some time for that. WDYT?

Reply:
Works for me, it is an implementation detail we can sort out later. We just need to make sure we can still extend JAX-RS API to add any missing behavior we need (e.g.: same site).

Reply:
@pedroigor +1